### ISSUE TITLE
[BUG] [Next 15.0.4 canary] Internal framework (rethrown) errors are broken

### ISSUE THREAD

**Issue by MrOxMasTer** [TheEdoRan/next-safe-action#299]
### Are you using the latest version of this library?

- [X] I verified that the issue exists in the latest next-safe-action release

### Is there an existing issue for this?

- [X] I have searched the existing issues and found nothing that matches

### Describe the bug

I have no idea what it's about

### Reproduction steps

1. Try entering mail in the field with unstable_cache
2. Try to enter mail again in the field with unstable_cache and look at the logs (make sure there are no errors)
3. Try to enter the mail again in the “use cache” field and make sure that the error pops up.

I tested: it happens even without ``use cache`` or `unstable_cache``. It happens in `canary` because it does not happen in `15.0.3`:

```
[ Server ] Error: (0 , next_dist_client_components_not_found_js__WEBPACK_IMPORTED_MODULE_0__.isNotFoundError) is not a function
```

### Expected behavior

no Error

### Link to a minimal reproduction of the issue

https://codesandbox.io/p/devbox/keen-cloud-gggnf4

### Operating System

Windows 11

### Library version

7.9.9

### Next.js version

15.0.4-canary.23-24

### Node.js version

22.10.0

**Comment by LouisCuvelier** [comment]
I'm getting the same kind of message :

```
Internal error: TypeError: (0 , next_dist_client_components_not_found_js__WEBPACK_IMPORTED_MODULE_0__.isNotFoundError) is not a function at executeMiddlewareStack
```

**Comment by MrOxMasTer** [comment]
```
timestamp=2024-12-01T19:30:54.724Z level=ERROR fiber=#96 message={}
timestamp=2024-12-01T19:30:54.729Z level=ERROR fiber=#96 message={}
timestamp=2024-12-01T19:30:54.732Z level=ERROR fiber=#96 message={}
 ⨯ node_modules\.pnpm\next-safe-action@7.9.9_next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-_pggdoqi72ubjxucbt36c45hjkm\node_modules\next-safe-action\dist\index.mjs (502:49) @ executeMiddlewareStack       
 ⨯ Internal error: TypeError: (0 , next_dist_client_components_not_found_js__WEBPACK_IMPORTED_MODULE_0__.isNotFoundError) is not a function
    at executeMiddlewareStack (./node_modules/.pnpm/next-safe-action@7.9.9_next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-_pggdoqi72ubjxucbt36c45hjkm/node_modules/next-safe-action/dist/index.mjs:518:196)  
    at async eval (./node_modules/.pnpm/next-safe-action@7.9.9_next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-_pggdoqi72ubjxucbt36c45hjkm/node_modules/next-safe-action/dist/index.mjs:548:6)
    at async C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\compiled\next-server\app-page-experimental.runtime.dev.js:157:2088 
    at async handleAction (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\compiled\next-server\app-page-experimental.runtime.dev.js:156:20482)
    at async renderToHTMLOrFlightImpl (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\compiled\next-server\app-page-experimental.runtime.dev.js:161:22008)
    at async doRender (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:1571:34)
    at async responseGenerator (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:1819:28)
    at async DevServer.renderToResponseWithComponentsImpl (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:1829:28)
    at async DevServer.renderPageComponent (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:2251:24)      
    at async DevServer.renderToResponseImpl (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:2289:32)     
    at async DevServer.pipeImpl (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:961:25)
    at async NextNodeServer.handleCatchallRenderRequest (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\next-server.js:278:17)
    at async DevServer.handleRequestImpl (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\base-server.js:854:17)
    at async C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\dev\next-dev-server.js:372:20
    at async Span.traceAsyncFn (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\trace\trace.js:153:20)
    at async DevServer.handleRequest (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\dev\next-dev-server.js:369:24)     
    at async invokeRender (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\lib\router-server.js:183:21)
    at async handleRequest (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\lib\router-server.js:361:24)
    at async requestHandlerImpl (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\lib\router-server.js:385:13)
    at async Server.requestListener (C:\Users\oxmas\Desktop\petProjects\shopco\node_modules\.pnpm\next@15.0.4-canary.24_@babel+core@7.25.8_react-dom@19.0.0-rc-380f5d67-20241113_react@19.0.0-r_a4tzqh3r6ouihinqpquitzmgki\node_modules\next\dist\server\lib\start-server.js:144:13)
digest: "3114230114"
  500 |                                                 }
  501 |                                                 }
 catch (e) {
> 502 |                                                 if (isRedirectError(e) || isNotFoundError(e)) {
      |                                                  
                                        ^
  503 |                                                 middlewareResult.success = true;
  504 |                                                 frameworkError = e;
  505 |                                                 return;
 POST /auth 500 in 322ms
```

**Comment by typeofweb** [comment]
Temp solution is to downgrade to `next@15.0.4-canary.17`.

**Comment by hoangtrung99** [comment]
I encountered a similar issue with the latest version of Next.js today. `next@15.0.4-canary.45`
`[ Server ] Error: (0 , next_dist_client_components_redirect_js__WEBPACK_IMPORTED_MODULE_1__.isRedirectError) is not a function`

**Comment by TheEdoRan** [comment]
[v7.9.10-beta.1](https://github.com/TheEdoRan/next-safe-action/releases/tag/v7.9.10-beta.1) should solve these issues. However, due to changes in Next.js's internal error handling mechanism, the next stable release of this library will only be compatible starting from the upcoming Next.js version (presumably 15.0.5). I really don't want to introduce breaking changes, but there's little I can do about it, since the underlying framework APIs have changed a lot, and this library heavily relies on them.

Let me know if the beta version solves these issues for you as well using Next.js canary, thanks!

**Comment by niikeec** [comment]
@TheEdoRan issue solved, thanks for the update

**Comment by MrOxMasTer** [comment]
> [v7.9.10-beta.1](https://github.com/TheEdoRan/next-safe-action/releases/tag/v7.9.10-beta.1) should solve these issues. However, due to changes in Next.js's internal error handling mechanism, the next stable release of this library will only be compatible starting from the upcoming Next.js version (presumably 15.0.5). I really don't want to introduce breaking changes, but there's little I can do about it, since the underlying framework APIs have changed a lot, and this library heavily relies on them.
> 

https://nextjs.org/blog/next-15-1
I think it's possible to go from 15.0.5, not 15.1.0 anymore 
(P.S. 15.0.5 just doesn't exist).

**Comment by headironc** [comment]
Similar issue `next@15.1.0`

`[TypeError: (0 , __TURBOPACK__imported__module__$5b$project$5d2f$node_modules$2f2e$pnpm$2f$next$40$15$2e$1$2e$0_react$2d$dom$40$19$2e$0$2e$0_react$40$19$2e$0$2e$0_$5f$react$40$19$2e$0$2e$0$2f$node_modules$2f$next$2f$dist$2f$client$2f$components$2f$redirect$2e$js__$5b$app$2d$rsc$5d$__$28$ecmascript$29$__.isRedirectError) is not a function] {
  digest: '70449113'
}`

**Comment by TheEdoRan** [comment]
@headironc Yep, 15.0.4-canary has been released as stable 15.1.

**Comment by rPr3v0t** [comment]
Same issue  here with latest version of 15.2.1-canary.1 and next-safe-action 10.4.. and i'm little stuck here because i have to use experimental Middleware on nodejs from 15.2.1-canary.1 because i use drizzle with Pg and i there is an issue with crypto (nodejs) with pg lib used by Drizzle Pg Adapter... (not compatible with next middleware on the edge in previous next version)

Middleware in nodejs in 15.2.1-canary.1 save my life XD so how can i fix the [ Server ] Error: (0 , {imported module [project]/nodemodules/.pnpm/next@15.1.0react-dom@19.0.0react@19.0.0__react@19.0.0/nodemodules/next/dist/client/components/redirect.js [app-rsc] (ecmascript)}.isRedirectError) is not a function with 15.2.1-canary.1 ? 

Thanks
